### PR TITLE
feat(cu): add util for getting chain ID

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds a one-liner for getting chain ID given an ethers.Provider input.
Going to follow this PR up with a good first issue to use this util
wherever we're trying to get the chain ID.

### CODE REVIEW

File: packages/core-utils/src/external/ethers/network.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Provider } from '@ethersproject/abstract-provider'
+
+export const getChainId = async (provider: Provider): Promise<number> => {
+  const network = await provider.getNetwork()
+  return network.chainId
```
Comment (roninjin10) [inline packages/core-utils/src/external/ethers/network.ts:5]: You could consider returning an entire object.  What I recently added was a common interface to use for everything.

interface Chain {
  id: number
  chainName: string
  isTestChain: boolean
  isLocalChain: boolean
  multicallAddress?: string
  getExplorerAddressLink: (string) => string
  getExplorerTransactionLink: (string) => string
  isL1: boolean
  isL2: boolean
  get l1() { // returns a Chain object of the corresponding L1 in the L1 -L2 pairing }
  get l2() { // analagous to get l1() but I imagine in future it returns an array of multiple l2s}
}

**Review by roninjin10 — APPROVED** [review]
lgtm.  Added a suggestion if you thought it was a good idea for your use case.

File: packages/core-utils/src/external/ethers/network.ts (hunk 1 above)
Comment (smartcontracts) [reply packages/core-utils/src/external/ethers/network.ts:5]: This is a cool idea, will do in a separate PR

Comment (roninjin10) [reply packages/core-utils/src/external/ethers/network.ts:5]: Yea the get l1() and get l2() on the frontend specifically really simplifies our code a ton where we get rid of branching like

const l2Balance = getBalance({chain: chainId === 1 ? 10 : chainId === kovan ? kovanOpt : chainId === 10 ? 10 : ...})

const l2Balance = getBalance({chain: chain.l2.id})

**Review by Inphi — APPROVED** [review]

**Pull request merged**